components/03-modules/popup/popup.js

```diff
@@ -1,33 +1,40 @@
 (function() {
 
   'use strict';
-
-  const popupsButtons = [...document.querySelectorAll('.popup__button')];
+  const popupsButtons = [...document.querySelectorAll('.popup-trigger')],
+        popupSelector = '.popup',
+        popupContentSelector = '.popup__content';
 
   class Popup {
-    constructor(button, popup = '.popup') {
+    constructor(button, popup = popupSelector) {
       this.button = button;
-      this.popup = this.button.parentNode.querySelector(popup);
-
-      this.body = document.querySelector('body');
-
+      this.buttonId = button.dataset.popuptrigger;
+      this.popup = document.querySelector(`${popup}[data-popup=${this.buttonId}]`);
+      this.popupContent = this.popup.querySelector(popupContentSelector);
       this.active = 'popup--active';
-
-      this.arrow = document.createElement('div');
-      this.arrow.classList.add('popup__arrow');
-      this.arrowSize = 21;
-
-      this.popup.appendChild(this.arrow);
-      this.arrowSelector = this.popup.querySelector('.popup__arrow');
+      this.body = document.querySelector('body');
+      this.closeButton = this.popup.querySelector('.popup__close-btn');
```

Comment:
can we use here alph align or some other sorting? And maybe some formatting

Reply:
formatting added and sorting applied